### DISCUSSION TITLE
Improvements and additions to Sound Blaster 16/AWE32/SB32 PnP DSP emulation

### DISCUSSION THREAD

**Discussion by Ompronce** [86Box/86Box#3476] [Ideas]
**Is your feature request related to a problem? Please describe.**
On the Vogons forum, I found a thread where a user linked to a dump of a Sound Blaster 16 DSP MCU (8051/8052), version 4.13, which is a ZIP file containing the dump of the MCU ROM and XOR tables.

**Describe the solution you'd like**
This could be useful for improving the DSP emulation of the Sound Blaster 16, AWE32, and SB32 PnP cards (this would also work for SB16, AWE32 and SB32 non-PnP cards), which could allow a DSP emulation that is closer to the the real hardware. A contributor with more advanced programming experience could use this information to improve the emulation of the Sound Blaster 16 or AWE32 DSPs.

**Describe alternatives you've considered**
I haven't considered or seen any alternative implementation as of the time of writing.

**Additional context**
The Vogons thread containing information about the DSP dump and more information is linked here:

https://www.vogons.org/viewtopic.php?p=621750#p621750

This feature appears to be emulated in MAME as well, and a link to the code from MAME which references this is linked here:

https://github.com/mamedev/mame/blob/master/src/devices/bus/isa/sb16.cpp

A ZIP file containing the DSP dump has been attached to this feature request as well.

[sb16.zip](https://github.com/86Box/86Box/files/6952317/sb16.zip)

**Comment by OBattler** [comment]
It is currently not possible, as it would require the emulation of all the bus signals, which we currently do not emulate.
